modules/react/text-input/lib/InputGroup.tsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+
+import {space} from '@workday/canvas-kit-react/tokens';
+import {createComponent, ExtractProps, useIsRTL} from '@workday/canvas-kit-react/common';
+import {Flex} from '@workday/canvas-kit-react/layout';
+
+import {TextInput} from './TextInput';
+
+export const InputGroupInnerStart = createComponent('div')({
+  Component(elemProps: ExtractProps<typeof Flex>, ref, Element) {
+    return (
+      <Flex
+        ref={ref}
+        as={Element}
+        position="absolute"
+        alignItems="center"
+        justifyContent="center"
+        height="xl"
+        width="xl"
+        {...elemProps}
+      />
+    );
+  },
+});
+
+export const InputGroupInnerEnd = createComponent('div')({
+  Component(elemProps: ExtractProps<typeof Flex>, ref, Element) {
+    return (
+      <Flex
+        ref={ref}
+        as={Element}
+        position="absolute"
+        alignItems="center"
+        justifyContent="center"
+        height="xl"
+        width="xl"
+        {...elemProps}
+      />
+    );
+  },
+});
+
+export const InputGroupInput = createComponent(TextInput)({
+  Component(elemProps: ExtractProps<typeof Flex>, ref, Element) {
+    return <Flex ref={ref} as={Element} width="100%" {...elemProps} />;
+  },
+});
+
+// make sure we always use pixels if the input is a number - this is required for `calc`
+const toPx = (input: string | number): string => {
+  return typeof input === 'number' ? `${input}px` : input;
+};
+
+// wrap an array of widths into something the browser can understand, including `calc` for multiple
+// values
+const wrapInCalc = (values: (string | number)[]): string | number | undefined => {
+  if (values.length === 0) {
+    return undefined;
+  }
+  if (values.length === 1) {
+    return values[0];
+  }
+  return `calc(${values.map(toPx).join(' + ')})`;
+};
+
+/**
+ * An `InputGroup` is a container around a {@link TextInput} with optional start and end elements.
```

Comment:
```suggestion
 * An `InputGroup` is a container around a {@link TextInput} with optional inner start and inner end elements.
```

Reply:
no idea if that suggestion makes sense, but trying to be consistent with the component name